handler/artifact_upload_handler.py

```diff
@@ -0,0 +1,114 @@
+"""
+A tornado application to handle uploading a conda build artifact to a specified
+file location.
+
+"""
+
+import argparse
+import hashlib
+import hmac
+import os
+import subprocess
+
+import tornado.httpserver
+import tornado.ioloop
+import tornado.web
+
+
+class ArtifactUploadHandler(tornado.web.RequestHandler):
+    """Handle writing conda build artifacts to a channel."""
+
+    def initialize(self, write_path, conda_exe, hash_expected):
+        """
+        "Hook for subclass initialization."
+
+        In this class, when we initialize `ArtifactUploadHandler`, we need to
+        set:
+         * the path that incoming files will be written to,
+         * the path to the conda executable to use for indexing the channel,
+         * the expected hashed and salted digest of the secure upload token.
+
+        """
+        self.write_path = write_path
+        self.conda_exe = conda_exe
+        self.hash_expected = hash_expected
+
+    def _check_token(self, token):
+        """
+        Check we have been passed the *correct* secure token before uploading
+        the artifact to the channel.
+
+        """
+        salt = '42679ad04d44c96ed27470c02bfb28c3'
```

Comment:
@dkillick Are you sure this is implemented correctly? My understanding of salts is that they should be randomly generated for each use so that the same password (token in this case) can be reused without resulting in the same hash.

Reply:
I don't know about correctly, but it works. It certainly isn't best practice, but it was good enough as a starting point...

Improvements welcome!